### DISCUSSION TITLE
Share button should also be controlled via roles

### DISCUSSION THREAD

**Discussion by saurabnigam** [apache/superset#18468] [Ideas]
**Is your feature request related to a problem? Please describe.**
Share button such as dashboard or share chart should be optional 

**Describe the solution you'd like**
A role for showing the share button would be sufficient

**Describe alternatives you've considered**
Adding CSS class specific for share button so custom CSS could be written to remove the element if not required

**Additional context**
We made a dashboard and want to share it externally. Problem is that there is a share button in charts as well which is not required and there is no way of removing it

**Comment by junlincc** [comment]
some efforts have been made towards this goal. 
[[SIP-51] Dashboard Level Access](https://github.com/apache/superset/issues/10408)
https://github.com/apache/superset/pull/13108

cc @amitmiran137

**Comment by amitmiran137** [comment]
All other option on chart menu(explore /export and etc') are controlled via a permission that been been given to a role.

I think that share chart should also behave the same in the form of ' share chart permission that can be linked to any role.


Wdyt?

**Comment by amitmiran137** [comment]
@saurabnigam if you'll create a PR with the new permission for share chart I'll support you
